src/Avalonia.Controls.TreeDataGrid/Models/TreeDataGrid/HierarchicalRows.cs

```diff
@@ -141,6 +142,11 @@ public void UnrealizeCell(ICell cell, int rowIndex, int columnIndex)
             (cell as IDisposable)?.Dispose();
         }
 
+        public int GetParentRowIndex(IndexPath modelIndex)
+        {
+            return ModelIndexToRowIndex(new IndexPath(modelIndex.Take(modelIndex.Count - 1)));
```

Comment:
I think you can use the C# range operator here to get the parent path: `modelIndex[..^1]` - it's used elsewhere in the codebase.